cibuildwheel/options.py

```diff
@@ -150,25 +152,78 @@ class ConfigOptionError(KeyError):
     pass
 
 
-def _dig_first(*pairs: tuple[Mapping[str, Setting], str], ignore_empty: bool = False) -> Setting:
+class Inherit(enum.Enum):
+    NONE = enum.auto()
+    APPEND = enum.auto()
+    PREPEND = enum.auto()
+
+
+def _dig_first(
+    *pairs: tuple[Mapping[str, Setting], str, Inherit], ignore_empty: bool = False
+) -> Setting:
```

Comment:
This function seems like it is doing two things, I wonder if we can split it up a bit.